source/core/slang-performance-profiler.cpp

```diff
@@ -39,11 +41,53 @@ namespace Slang
             for (auto func : data)
             {
                 out << func.key << ": \t";
-                out << func.value.invocationCount << "\t" << func.value.duration.count() / 1000000.0f << "\n";
+                auto milliseconds = std::chrono::duration_cast< std::chrono::milliseconds >(func.value.duration);
+                out << func.value.invocationCount << "\t" << milliseconds.count() << "ms\n";
             }
         }
     };
 
+    class SerialPerformaceProfilerImpl : public SerialPerformaceProfiler
+    {
+    public:
+       void accumulateResults(StringBuilder& out, PerformanceProfiler* profile)
+       {
+           PerformanceProfilerImpl* profilerImpl = static_cast<PerformanceProfilerImpl*>(profile);
+            for (auto func : profilerImpl->data)
+            {
+                auto milliseconds = std::chrono::duration_cast< std::chrono::milliseconds >(func.value.duration);
+                long ms = milliseconds.count();
+
+                m_mutex.lock();
```

Comment:
I think this mutex_lock will be better go outside of the for-loop.
It will keep locking and unlocking too many times.

Reply:
done.